app/views/hosts/welcome.html.erb

```diff
@@ -8,6 +8,6 @@
   <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("4.4Provisioning")%></p>
   <div class="blank-slate-pf-main-action">
     <%= new_link(_('Create Host'), {}, { :class => "btn-lg" }) %>
-    <%= display_link_if_authorized(_("Register Host"), { :controller => 'registration', :action => :new }, :class => "btn btn-default btn-lg") %>
+    <%= display_link_if_authorized(_("Register Host"), hash_for_register_hosts_path, :class => "btn btn-default btn-lg") %>
```

Comment:
I'm sorry I've confused you, but I'm not sure this will work without named routes, so this will probably need to be part of https://projects.theforeman.org/issues/31849 IMO.

Reply:
Well it works on my setup ... can you give me the error you have?

Reply:
nope, I can't because it was just a hunch, Rails does not generate named routes for `get` automatically, but only if you use `as:` option, but if it works, then great and don't mind me! :100: